docs/decisions/20240115-Error-handling-report-problem-agent.md

```diff
@@ -0,0 +1,166 @@
+# Error Handling Report Problem for Agent
+
+- Status: draft
+- Deciders: Shailesh Patil, Fabio Pinheiro, Yurii Shynbuiev, Benjamin Voiturier, David Poltorak
+- Date: 2024-01-16
+- Tags: report, problem, error, handling, agent
+
+Technical Story: [Error Handling Report Problem for Agent]
+
+## Context and Problem Statement
+
+In decentralized systems like those with remote collaborating agents, effectively reporting errors and warnings is a complex task. 
+It's crucial to not only highlight problems but also provide the necessary context to those who need this information, 
+which might include different groups: those who should be informed and those who can actually resolve the issues. 
+
+It is more challenging when a problem is identified significantly later or in a different location from where it originated, 
+involving collaboration among various parties for a solution.
+For DIDComm, the aspect of interoperability is especially critical in the context of problem reporting. 
+In DIDComm, an agent developed by one team is required to be adept at interpreting errors reported by an agent from a completely different team,
+presenting a unique challenge in this area.
+
+In Agent Currently  we have 3 flows Connection Issuance and Verification.
+Communication between 2 agents Agent A <-> Agent B happens in the background job in async
```

Comment:
Can you speak more to the point about background jobs? As I know the architecture, it's clear what this means - but - for someone new - this may give them the wrong idea.

Something like - `DIDComm protocols and the tranistions between states in the flow sequence are processed asyncrhonously by recurring jobs that....`